### PR TITLE
Page Anchor Component > Unit Tests

### PR DESCRIPTION
**Pull request by Blackbaud-SteveBrush**

### CODE REVIEW

File: src/modules/page-anchor/page-anchor.component.ts [hunk 1]
```diff
@@ -22,9 +22,8 @@ export class StachePageAnchorComponent implements OnInit, StacheNavLink {
 
   public ngOnInit(): void {
     const element = this.elementRef.nativeElement;
-    let name = element.innerText || element.textContent;
-    name = name.trim();
-    let fragment = name.toLowerCase().replace(/ /g, '-').replace(/[^\w-]+/g, '');
+    let name = element.textContent;
+    let fragment = name.trim().toLowerCase().replace(/ /g, '-').replace(/[^\w-]+/g, '');
```
Comment (Blackbaud-BrandonJones) [inline src/modules/page-anchor/page-anchor.component.ts]: do we want this chain all on one line?

File: src/modules/page-anchor/page-anchor.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      .querySelector('.stache-page-anchor')
+      .getAttribute('id');
+    expect(id).toBe('test-content');
+  });
+
+  it('should call the addHashToUrl method when the heading is clicked', () => {
+    spyOn(component, 'addHashToUrl');
+    const heading = debugElement.nativeElement.querySelector('.stache-page-anchor-heading');
+    heading.click();
+    fixture.whenStable().then(() => {
```
Comment (Blackbaud-BrandonJones) [inline src/modules/page-anchor/page-anchor.component.spec.ts]: .then to the next line.

File: src/modules/page-anchor/page-anchor.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+      expect(component.addHashToUrl).toHaveBeenCalled();
+    });
+  });
+
+  it('should call scrollIntoView on the heading when clicked', () => {
+    spyOn(debugElement.nativeElement, 'scrollIntoView');
+    component.addHashToUrl();
+    fixture.whenStable().then(() => {
```
Comment (Blackbaud-BrandonJones) [inline src/modules/page-anchor/page-anchor.component.spec.ts]: same here.

File: src/modules/page-anchor/page-anchor.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    spyOn(component, 'addHashToUrl');
+    const heading = debugElement.nativeElement.querySelector('.stache-page-anchor-heading');
+    heading.click();
+    fixture.whenStable().then(() => {
+      expect(component.addHashToUrl).toHaveBeenCalled();
+    });
+  });
+
+  it('should call scrollIntoView on the heading when clicked', () => {
+    spyOn(debugElement.nativeElement, 'scrollIntoView');
```
Comment (blackbaud-brandonhare) [inline src/modules/page-anchor/page-anchor.component.spec.ts]: I'm guessing this bubbles up? Go figure. Just glad it works.

File: src/modules/page-anchor/page-anchor.component.spec.ts (hunk 2 above)
Comment (Blackbaud-SteveBrush) [inline src/modules/page-anchor/page-anchor.component.spec.ts]: If we do it here, we gotta do it on all the specs...

Comment (Blackbaud-BrandonJones) [inline src/modules/page-anchor/page-anchor.component.spec.ts]: I must not have seen them in the other specs so far?   I thought our convention was always dangle the second and subsequent chained methods....?

File: src/modules/page-anchor/page-anchor.component.spec.ts (hunk 4 above)
Comment (Blackbaud-SteveBrush) [inline src/modules/page-anchor/page-anchor.component.spec.ts]: I believe the `debugElement` is synonymous to the `this.elementRef` in the component, so they're the same thing.

**Comment by Blackbaud-BrandonJones** [comment]
Looks good!

**Pull request merged**